wicket/src/rack_setup.rs

```diff
@@ -0,0 +1,254 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Support for rack setup configuration via wicketd.
+
+use crate::wicketd::create_wicketd_client;
+use anyhow::bail;
+use anyhow::Context;
+use anyhow::Result;
+use clap::Subcommand;
+use omicron_passwords::Password;
+use omicron_passwords::PasswordHashString;
+use slog::Logger;
+use std::io;
+use std::io::Read;
+use std::mem;
+use std::net::SocketAddrV6;
+use std::time::Duration;
+use wicket_common::rack_setup::PutRssUserConfigInsensitive;
+use wicketd_client::types::CertificateUploadResponse;
+use wicketd_client::types::NewPasswordHash;
+use wicketd_client::types::PutRssRecoveryUserPasswordHash;
+use zeroize::Zeroizing;
+
+mod config_toml;
+
+use config_toml::TomlTemplate;
+
+const WICKETD_TIMEOUT: Duration = Duration::from_secs(5);
+
+#[derive(Debug, Subcommand)]
+pub(crate) enum SetupArgs {
+    /// Get the current rack configuration as a TOML template
+    ///
+    /// Save this template to a file, edit it, then upload it via `set-config`.
+    GetConfig,
+
+    /// Set the current rack configuration from a filled-in TOML template
+    SetConfig,
+
+    /// Reset the configuration to its original (empty) state
+    ResetConfig,
```

Comment:
To confirm, this also resets the password, certificate and key right? If so, worth adding some documentation about this here.